### PR TITLE
Update vfs_zfsacl.c - proposed fix for #26039

### PR DESCRIPTION
**Pull request by anodos325**
Set SEC_DESC_DACL_PROTECTED if no ACEs in ACL have "inherited" flag. This makes it so that the "Disable Inheritance" -> "Convert inherited permissions into explicit permissions on this object" button in Windows File Explorer behaves correctly.

This was tested against Samba 4.6.8 in net/samba46 port in FreeBSD.

### CODE REVIEW

**Comment by anodos325** [comment]
Note: this patch requires patching sysutils/libsunacl to support "inherited" flag.

**Comment by kmoore134** [comment]
Jenkins: add to whitelist

**Comment by kmoore134** [comment]
Jenkins: retest this please

File: source3/modules/vfs_zfsacl.c [hunk 1]
```diff
@@ -52,6 +52,7 @@ static NTSTATUS zfs_get_nt_acl_common(struct connection_struct *conn,
 	const SMB_STRUCT_STAT *psbuf = NULL;
 	int ret;
 	bool is_dir;
+	uint16_t inherited_present;
```
Comment (b-a-t) [inline source3/modules/vfs_zfsacl.c]: Andrew, again, for the flag it's more sane to use:
```
bool is_inherited;
```

**Review by jhixson74 — APPROVED** [review]

**Comment by anodos325** [comment]
Jenkins: retest this please

**Pull request merged**